pygeohash/nbgeohash.py

```diff
@@ -0,0 +1,186 @@
+"""
+.. module:: nbgeohash
+   :platform: Unix, Windows
+   :synopsis: A module for encoding to and decoding from the geohash system
+
+.. moduleauthor:: ilyas Moutawwakil <ilyas.moutawwakil@gmail.com>
+
+
+"""
+
+
+from math import log10
+
+__base32 = "0123456789bcdefghjkmnpqrstuvwxyz"
+#  Note: the alphabet in geohash differs from the common base32 alphabet described in IETF's RFC 4648
+# (http://tools.ietf.org/html/rfc4648)
+
+try:
+    # Soft dependency
+    import numpy as np
+    from numba import njit, types
+except ImportError:
+    print("Numpy and Numba are soft dependencies to use this feature.")
+    raise ImportError(
+        "Couldn't import numpy or numba, make sure they are installed properly."
+    )
+
+__author__ = "Ilyas Moutawwakil"
+
+
+@njit(cache=True, fastmath=True)
```

Comment:
is there a way to catch errors thrown here in the case of numba not being present? Maybe we import njit if numba is installed, and define a dummy decorator that just no-ops and raises an import error also called njit if numba is not installed?